Reference operator in `network_time.h` header file

**Describe the bug**
I'm not sure wether it's desired, but it looks like in the `network_time.h` a reference operator has sneaked in:
```c
otNetworkTimeStatus otNetworkTimeGet(otInstance *aInstance, uint64_t &aNetworkTime);
```
This results in a build error when trying to include this file from some other `.c` file.

**To Reproduce**
Information to reproduce the behavior, including:
1. Git commit id: 0208d10563aa21c518092985c78ecf9cd223ab74
2. IEEE 802.15.4 hardware platform: nRF52840
3. Build steps: 
For the purposes of this example, I simply added `#include <openthread/network_time.h>
` to the `examples/apps/cli/main.c`. Then I executed
```
./bootstrap
make -f ./examples/Makefile-nrf52840
```
4. Network topology - not relevant I think

**Expected behavior**
I would like to use network time synchronization functions from OT API in the `.c` source files. 

**Console/log output**
```
In file included from openthread/examples/../examples/apps/cli/main.c:37:0:
openthread/examples/../include/openthread/network_time.h:87:70: error: expected ';', ',' or ')' before '&' token
 otNetworkTimeStatus otNetworkTimeGet(otInstance *aInstance, uint64_t &aNetworkTime);
                                                                      ^
In file included from openthread/examples/../examples/apps/cli/main.c:37:0:
openthread/examples/../include/openthread/network_time.h:87:70: error: expected ';', ',' or ')' before '&' token
 otNetworkTimeStatus otNetworkTimeGet(otInstance *aInstance, uint64_t &aNetworkTime);
```

--- a/include/openthread/network_time.h
+++ b/include/openthread/network_time.h
@@ -84,7 +84,7 @@ typedef void (*otNetworkTimeSyncCallbackFn)(void *aCallbackContext);
  * @returns The time synchronization status.
  *
  */
-otNetworkTimeStatus otNetworkTimeGet(otInstance *aInstance, uint64_t &aNetworkTime);
+otNetworkTimeStatus otNetworkTimeGet(otInstance *aInstance, uint64_t *aNetworkTime);
 
 /**
  * Set the time synchronization period.

--- a/src/cli/cli.cpp
+++ b/src/cli/cli.cpp
@@ -1870,7 +1870,7 @@ void Interpreter::ProcessNetworkTime(int argc, char *argv[])
         uint64_t            time;
         otNetworkTimeStatus networkTimeStatus;
 
-        networkTimeStatus = otNetworkTimeGet(mInstance, time);
+        networkTimeStatus = otNetworkTimeGet(mInstance, &time);
 
         mServer->OutputFormat("Network Time:     %luus", time);
 

--- a/src/core/api/network_time_api.cpp
+++ b/src/core/api/network_time_api.cpp
@@ -42,11 +42,11 @@
 
 using namespace ot;
 
-otNetworkTimeStatus otNetworkTimeGet(otInstance *aInstance, uint64_t &aNetworkTime)
+otNetworkTimeStatus otNetworkTimeGet(otInstance *aInstance, uint64_t *aNetworkTime)
 {
     Instance &instance = *static_cast<Instance *>(aInstance);
 
-    return instance.Get<TimeSync>().GetTime(aNetworkTime);
+    return instance.Get<TimeSync>().GetTime(*aNetworkTime);
 }
 
 otError otNetworkTimeSetSyncPeriod(otInstance *aInstance, uint16_t aTimeSyncPeriod)

--- a/src/ncp/ncp_base_mtd.cpp
+++ b/src/ncp/ncp_base_mtd.cpp
@@ -3182,7 +3182,7 @@ template <> otError NcpBase::HandlePropertyGet<SPINEL_PROP_THREAD_NETWORK_TIME>(
     otNetworkTimeStatus networkTimeStatus;
     uint64_t            time;
 
-    networkTimeStatus = otNetworkTimeGet(mInstance, time);
+    networkTimeStatus = otNetworkTimeGet(mInstance, &time);
 
     SuccessOrExit(error = mEncoder.WriteUint64(time));
     SuccessOrExit(error = mEncoder.WriteInt8((int8_t)networkTimeStatus));

